### PR TITLE
Add fuzz testing for PGWire connections

### PR DESCRIPTION
**Pull request by nvb**
This change adds support for using `go-fuzz` to fuzz test CockroachDB packages which depend on C++ files. Using this new functionality, it adds a test to fuzz PGWire v3Connections.

### CODE REVIEW

File: go-fuzz-build/main.go [hunk 1]
```diff
@@ -231,13 +231,28 @@ func createFuzzMain(pkg string) {
...
+
 func clonePackage(workdir, pkg, targetPkg string) {
 	dir := goListProps(pkg, "Dir")[0]
 	if !strings.HasSuffix(filepath.ToSlash(dir), pkg) {
 		failf("package dir '%v' does not end with import path '%v'", dir, pkg)
 	}
 	newDir := filepath.Join(workdir, "src", targetPkg)
-	copyDir(dir, newDir, false, isSourceFile)
+	rec := false
+	if _, ok := recPkgSet[pkg]; ok {
```
Comment (tamird) [inline go-fuzz-build/main.go]: isn't this the same as `_, rec := recPkgSet[pkg]`?

Comment (nvb) [inline go-fuzz-build/main.go]: Done.

**Comment by petermattis** [comment]
Nice. Fuzz testing pgwire was on my wish list. Have we revisited fuzz testing sql parsing recently?

**Comment by nvb** [comment]
I haven't looked into it in a while, but last time I did (~3 months ago), I ran fuzzing for about a day with no new crashes. It sounds like it's worth another look though with all the new parser changes. I'll investigate.

**Pull request merged**